R/dash.R

```diff
@@ -772,17 +775,37 @@ Dash <- R6::R6Class(
     # ------------------------------------------------------------------------
     callback = function(output, params, func) {
       assert_valid_callbacks(output, params, func)
-
+      
       inputs <- params[vapply(params, function(x) 'input' %in% attr(x, "class"), FUN.VALUE=logical(1))]
       state <- params[vapply(params, function(x) 'state' %in% attr(x, "class"), FUN.VALUE=logical(1))]
-
+      
       if (is.function(func)) {
         clientside_function <- NULL
+      } else if (is.character(func)) {
+        # update the scripts before generating tags, and remove exact
+        # duplicates from inline_scripts
+        fn_name <- paste0("_dashprivate_", output$id)
+
+        func <- paste0('<script>\n',
+                       '                var clientside = window.dash_clientside = window.dash_clientside || {};\n',
```

Comment:
Maybe @Marc-Andre-Rivet can weigh in, but should we begin to move towards `let`'s and `const`'s instead of 'var`'s for "modern" JS syntax? Although on the other hand, if we want to match the Python syntax it makes more sense to stick with `var` for now.

Reply:
This code is inserted directly on the page, it's not transpiled. So unfortunately we can't safely use modern syntax here.

Reply:
non-blocking, but the leading whitespace in this string is unnecessary. On the Python side, the results end up looking something like:
```js
<script>
var clientside = window.dash_clientside = window.dash_clientside || {};
var ns = clientside["_dashprivate_clientside-graph"] = clientside["_dashprivate_clientside-graph"] || {};
ns["figure"] = 
    function(data, scale) {
        return {
            'data': data,
            'layout': {
                 'yaxis': {'type': scale}
             }
        }
    }
    ;
</script>
```
But almost all the whitespace actually came from the user input. In fact if we ever decide we want to optimize bytes we can strip out all leading whitespace and carriage returns from our code and at least leading whitespace from the user-provided function string.

Reply:
fixed in 09c73a2 and 989643a

Reply:
I'll contribute the commit hook in a separate PR; want to test a little bit beforehand.